### ISSUE TITLE
[QoL] Woof!-like weapon switching

### ISSUE THREAD

**Issue by JustinWayland** [Helion-Engine/Helion#886]
This is just a QoL thing, but something in the vein of Woof!'s weapon slots from the 15.0.0 version would be very nice to have.

While this was not what Woof! went with, it illustrates the idea well enough:

![359753703-e76c12e7-918c-4240-8ef6-00318adcd689](https://github.com/user-attachments/assets/8535b641-619a-4b17-b966-97da9c49cdd1)

**Comment by lemming104** [comment]
I'm going to punt this one to @nstlaurent for if he chooses to add some sort of weapon grouping construct.  We could then map the D-pad to groups 1, 2, 3, 4.

**Comment by JustinWayland** [comment]
I was thinking about this the other day, and I realized that given the current architecture I might be able to get away with three new variables and a new command.

First, the variables
```
private int[,] GroupSlots = new int[4,3];
private int CurrentGroup = -1;
private int CurrentGroupIndex = -1;
```

Then, the new command: "use_weapon_group".  This would in essence select a weapon from a group according to the following procedure done with temporary versions of ```CurrentGroup``` and ```CurrentGroupIndex```:
1. Are any of the slots in the chosen group usable?  If not, do nothing.
2. Is the group we chose different from the group in ```CurrentGroup```? If so, then set the temporary version of ```CurrentGroupIndex``` to zero. Otherwise, change nothing.
3. Search the selected group for the first entry after the current entry that can be used, if any. Only use earlier entries if there is not a later entry, and keep to the current one only if no other usable entries can be found. Write the result to ```CurrentGroupIndex```.
4. Issue a "switch to weapon slot" command.

None of this seems out of my wheelhouse except for my lacking codebase knowledge.  I would gladly do the work myself, if I knew where to put the variables and commands and how to issue a switch to a numbered slot command.
